examples/platform/silabs/uart.cpp

```diff
@@ -542,18 +545,21 @@ void uartMainLoop(void * args)
 /**
  * @brief Send Bytes to UART. This blocks the UART task.
  *
- * @param buffer pointer to the buffer containing the data
- * @param nbOfBytes number of bytes to send
+ * @param buffer pointer to the UartTxStruct_t containing the data
  */
-void uartSendBytes(uint8_t * buffer, uint16_t nbOfBytes)
+void uartSendBytes(UartTxStruct_t * bufferStruct)
 {
 #if SLI_SI91X_MCU_INTERFACE
     // ensuring null termination of buffer
-    if (nbOfBytes != CHIP_SHELL_MAX_LINE_SIZE && buffer[nbOfBytes - 1] != '\0')
+    if (bufferStruct->length < ArraySize(bufferStruct->data) && bufferStruct->data[bufferStruct->length - 1] != '\0')
+    {
+        bufferStruct->data[bufferStruct->length] = '\0';
+    }
+    else
     {
-        buffer[nbOfBytes] = '\0';
+        bufferStruct->data[ArraySize(bufferStruct->data) - 1] = '\0';
     }
-    Board_UARTPutSTR(reinterpret_cast<uint8_t *>(buffer));
+    Board_UARTPutSTR(reinterpret_cast<uint8_t *>(bufferStruct->data));
```

Comment:
isn't `bufferStruct->data` already a uint8_t* ?

Reply:
Yes it is